### PR TITLE
made a bunch of optimizations

### PR DESCRIPTION
**Pull request by rickyelopez**
addresses issue #4 among other things with no issues

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -192,16 +192,19 @@ void loop(){
...
-	sw1 	=	!digitalRead(PB8);
-	sw2 	=	!digitalRead(PB6);
-	sw3 	=	!digitalRead(PB4);
-	sw4 	=	!digitalRead(PB3);
-	pl 		=	!digitalRead(PA8);
-	pr 		=	!digitalRead(PA9);
-	tl 		=	!digitalRead(PA10);
-	tr		=	!digitalRead(PA15);
-	
+	uint32_t portb = GPIOB->regs->IDR;
```
Comment (rickyelopez) [inline src/main.cpp:185]: direct port access instead of multiple digitalReads

File: src/main.cpp [hunk 2]
```diff
@@ -230,21 +234,25 @@ void get_inputs(uint8_t *status1, uint8_t *status2){
...
 	}
 
+	// un-dismiss the warnings if the middle button is pressed
 	dismissed = mid ? 0 : dismissed;
 
+	// create the status bitfields
 	*status1 = 0;
 	*status2 = 0;
 
-	*status1 = sw1 ? *status1 | sw1_st : *status1;
```
Comment (rickyelopez) [inline src/main.cpp:238]: turns out there's a simpler way to do this (see below)

**Comment by johnstonstott** [comment]
Builds successfully for me, code style seems consistent, optimizations and reorganizations are good. Is there anything else in particular that you wanted me to look over @rickyelopez?

**Comment by rickyelopez** [comment]
Ok sounds good, if you don't see any issues go ahead and approve!

**Review by johnstonstott — APPROVED** [review]

**Pull request merged**